samcli/lib/deploy/deployer.py

```diff
@@ -69,7 +69,7 @@ def has_stack(self, stack_name):
         """
         try:
             resp = self._client.describe_stacks(StackName=stack_name)
-            if len(resp["Stacks"]) != 1:
+            if not resp["Stacks"]:
```

Comment:
Does this handle an empty array?

Reply:
Actually, better question, can we ensure that we test the case of no such stack? Just to be sure for regressions.